From 9f9273f967bf2d0fa848a115d588ed4bfb629aac Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Wed, 3 Oct 2018 23:03:18 -0500
Subject: [PATCH 11/12] Static cast size_t to int in arguments 1,2 to
 forward_input_or_allocate_output()

This fix resolves the following compiler error:

tensorflow/core/kernels/mkl_relu_op.cc(1028): error C2398: Element '1':
conversion from 'const std::size_t' to 'int' requires a narrowing conversion
---
 tensorflow/core/kernels/mkl_relu_op.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tensorflow/core/kernels/mkl_relu_op.cc b/tensorflow/core/kernels/mkl_relu_op.cc
index f4cfc48..1018dcc 100644
--- a/tensorflow/core/kernels/mkl_relu_op.cc
+++ b/tensorflow/core/kernels/mkl_relu_op.cc
@@ -1024,7 +1024,8 @@ class MklReluGradOpBase : public OpKernel {
       }
 
       OP_REQUIRES_OK(context, context->forward_input_or_allocate_output(
-                                  {diff_dst_index}, diff_src_index,
+                                  {static_cast<const int>(diff_dst_index)},
+                                  static_cast<const int>(diff_src_index),
                                   tf_shape_diff_src, &diff_src_tensor));
       AllocateOutputSetMklShape(context, diff_src_index, dnn_shape_diff_src);
 
-- 
1.8.3.1

